src/testHelpers/withTimeMachine.jsx

```diff
@@ -1,21 +1,22 @@
-import React, { useEffect } from 'react';
+import { useEffect, useState } from 'react';
 import { element } from 'prop-types';
 import {
   startTimeMachine,
   resetTimeMachine,
 } from '../../.storybook/time-machine';
 // This affects the global Date object for the storybook application, to ensure consistency in chromaticQA testing.
 const WithTimeMachine = ({ children }) => {
-  // set time, but leave it ticking, so knobs work
-  startTimeMachine(false);
-  // setting tick to false, prevents the timemachine from ticking,
-  // so the timestamp displayed is frozen, and there are no side effects.
+  const [componentToRender, setComponentToRender] = useState(null);
+
   useEffect(() => {
-    resetTimeMachine();
-    startTimeMachine(true);
-    return resetTimeMachine();
+    function handleStateChange() {
+      setComponentToRender(children);
+    }
+    startTimeMachine();
+    handleStateChange();
+    return () => resetTimeMachine();
```

Comment:
I think we should add a comment explaining why this block is needed. I don't quite understand what's happening here 😅

Reply:
In this block we're basically starting the time-machine, rendering the component, then resetting the time-machine.

In the previous implementation, the time-machine was not being reset after the component renders. That's what was making it to mis-behave.

Reply:
Thanks for the explanation.